pkg/ocm/helpers.go

```diff
@@ -636,6 +638,49 @@ func ValidateSubnetsCount(multiAZ bool, privateLink bool, subnetsInputCount int)
 	return nil
 }
 
+func ValidateHostedClusterSubnets(awsClient aws.Client, isPrivate bool, subnetIDs []string) (int, error) {
+	if isPrivate && len(subnetIDs) < 1 {
+		return 0, fmt.Errorf("The number of subnets for a private hosted cluster should be at least one")
+	}
+	if !isPrivate && len(subnetIDs) < 2 {
+		return 0, fmt.Errorf("The number of subnets for a public hosted cluster should be at least two")
```

Comment:
Q: Is this because it needs at least one private + at least one public?  
However, I don't see privateSubnetCount >= 1 enforced anywhere.

These quick len() checks avoid the need for API calls, but perhaps the returned error is less informative than user could get if you proceeded anyway? :thinking: